flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/utils/SortSpec.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.nodes.exec.utils;
+
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.RowType;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/** SortSpec describes how the data will be sort. */
+public class SortSpec {
+
+    /** SortSpec does not require sort. */
+    public static final SortSpec ANY = new SortSpec(new SortFieldSpec[0]);
+
+    private final SortFieldSpec[] fieldSpecs;
+
+    public SortSpec(SortFieldSpec[] fieldSpecs) {
+        this.fieldSpecs = fieldSpecs;
+    }
+
+    public SortSpec getSubSpec(int startIndex) {
+        return new SortSpec(
+                Arrays.stream(fieldSpecs, startIndex, fieldSpecs.length)
+                        .toArray(SortFieldSpec[]::new));
+    }
```

Comment:
This method looks a little specific,  I would like to add a method named `get SortFieldSpecs`, and pick the target in specific class.

Reply:
I think it is ok to have some helper method for common use case.